src/Compiler/Service/IncrementalBuild.fs

```diff
@@ -358,26 +358,23 @@ type BoundModel private (tcConfig: TcConfig,
         let hasSig = this.BackingSignature.IsSome
 
         // If partial checking is enabled and we have a backing sig file, then use the partial state. The partial state contains the sig state.
-        if tcInfoNode.HasFull && enablePartialTypeChecking && hasSig then
-            let newTcInfoStateOpt =
-                match tcInfoNode with
-                | TcInfoNode(_, fullGraphNode) -> 
-                    let tcInfo, _ = fullGraphNode.TryPeekValue().Value
-                    Some(PartialState tcInfo)
-
-            BoundModel(
-                tcConfig,
-                tcGlobals,
-                tcImports,
-                keepAssemblyContents, keepAllBackgroundResolutions,
-                keepAllBackgroundSymbolUses,
-                enableBackgroundItemKeyStoreAndSemanticClassification,
-                enablePartialTypeChecking,
-                beforeFileChecked,
-                fileChecked,
-                prevTcInfo,
-                syntaxTreeOpt,
-                newTcInfoStateOpt)
+        if enablePartialTypeChecking && hasSig then
```

Comment:
```suggestion
match tcInfoNode.TryGetFull() with
| ValueSome (tcInfo, _) when enablePartialTypeChecking && hasSig ->
    BoundModel(
        tcConfig,
        tcGlobals,
        tcImports,
        keepAssemblyContents, keepAllBackgroundResolutions,
        keepAllBackgroundSymbolUses,
        enableBackgroundItemKeyStoreAndSemanticClassification,
        enablePartialTypeChecking,
        beforeFileChecked,
        fileChecked,
        prevTcInfo,
        syntaxTreeOpt,
        tcInfoStateOpt = Some (PartialState tcInfo))
| _ -> this
```

Idk how to properly do suggestions, but seems to me like this whole `if then else` can be replaced by this match now (unless we really don't want to call `TryGetFull` unless `enablePartialTypeChecking && hasSig` (I am not familiar with this part of the compiler at all)